internal/services/devcenter/dev_center_network_connection_resource_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package devcenter_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2023-04-01/networkconnections"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type DevCenterNetworkConnectionTestResource struct{}
+
+func TestAccDevCenterNetworkConnection_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dev_center_network_connection", "test")
+	r := DevCenterNetworkConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDevCenterNetworkConnection_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dev_center_network_connection", "test")
+	r := DevCenterNetworkConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccDevCenterNetworkConnection_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dev_center_network_connection", "test")
+	r := DevCenterNetworkConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("domain_password"),
+	})
+}
+
+func TestAccDevCenterNetworkConnection_update(t *testing.T) {
```

Comment:
shall we use `basic->complete->update->basic` to test add/update/remove properties?

Reply:
Updated